### PR TITLE
Simple defines

### PR DESCRIPTION
**Pull request by mkstoyanov**
* streamlined cmake variables, e.g., Tasmanian_ENABLE_CUDA is used in both cmake and c++ without the need for the intermediary TASMANIAN_CUDA; change also covers MPI and CUBLAS
* only python files are auto-generated for simple GNU-Make build, TasmanianConfig.hpp and tsgGetPaths.m are simple enough to maintain manually
* cached test threads and gpu variables and the MATLAB work folder; the behavior of the build engine has not changed, this makes things cleaner when using the cmake gui
* fixed some typos

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -82,6 +80,10 @@ option(Tasmanian_ENABLE_CUDA "Enable Nvidia GPU acceleration using CUDA kernels
 option(Tasmanian_STATIC_LIBRARY "Compile static library and the executables with static linkage (Tasmanian)" ON)
 option(Tasmanian_SHARED_LIBRARY "Compile dynamic library (Tasmanian)" ON)
 
+set(Tasmanian_TESTS_GPU_ID "-1" CACHE STRING "specify GPU ID for testing, -1 means running tests on all visiable devices")
```
Comment (dalg24) [inline CMakeLists.txt]: typo `s/visiable/visible`

File: Config/AltBuildSystems/TasmanianConfig.hpp [hunk 2]
```diff
@@ -31,9 +31,9 @@
 #ifndef __TASMANIAN_CONFIG_HPP
```
Comment (dalg24) [inline Config/AltBuildSystems/TasmanianConfig.hpp]: Should this file only be generated at the time of release?

File: Config/AltBuildSystems/TasmanianSG.py [hunk 3]
```diff
@@ -39,7 +39,7 @@
     bTsgPlotting = False
     tsgPlot = []
 
-__version__ = "5.1"
+__version__ = "6.0"
```
Comment (dalg24) [inline Config/AltBuildSystems/TasmanianSG.py]: Should this be coming from the `C/C++`?

File: Config/TasmanianConfig.in.hpp [hunk 4]
```diff
@@ -45,9 +45,9 @@
...
 ////////////////////////////////////////////////////////////////////////
 
-#cmakedefine TASMANIAN_CPU_BLAS
-#cmakedefine TASMANIAN_CUBLAS
-#cmakedefine TASMANIAN_CUDA
-#cmakedefine TASMANIAN_MPI
+#cmakedefine Tasmanian_ENABLE_BLAS
+#cmakedefine Tasmanian_ENABLE_CUBLAS
+#cmakedefine Tasmanian_ENABLE_CUDA
+#cmakedefine Tasmanian_ENABLE_MPI
```
Comment (dalg24) [inline Config/TasmanianConfig.in.hpp]: Sounds good.

File: Config/AltBuildSystems/TasmanianSG.py (hunk 3 above)
Comment (mkstoyanov) [reply Config/AltBuildSystems/TasmanianSG.py]: All version info comes from the same line in cmake, there is also regression test that ensures that those will match.

File: Config/TasmanianConfig.in.hpp (hunk 4 above)
Comment (mkstoyanov) [reply Config/TasmanianConfig.in.hpp]: Should have done this from the beginning.

**Pull request closed without merging**